Tests/scripts/hook_validations/structure.py

```diff
@@ -18,295 +29,281 @@
-                        "please update the file {}.".format(self.file_path))
-            self._is_valid = False
-
-        return self._is_valid
+        Returns:
+            bool. Whether the new dictionary is a sub set of the old dictionary.
+        """
+        for arg, required in old_dict.items():
+            if arg not in new_dict.keys():
+                return False
 
-    def is_valid_fromversion_on_modified(self, change_string=None):
-        """Check that the fromversion property was not changed on existing Content files.
+            if required != new_dict[arg] and new_dict[arg]:
+                return False
+
+        for arg, required in new_dict.items():
+            if arg not in old_dict.keys() and required:
+                return False
+        return True
+
+    @staticmethod
+    def get_file_id_from_loaded_file_data(loaded_file_data):
+        # type: (dict) -> Optional[str]
+        """Gets a dict and extracting its `id` field
 
         Args:
-            change_string (string): the string that indicates the changed done on the file(git diff)
+            loaded_file_data: Data to find dict
 
         Returns:
-            bool. Whether the files' fromversion as been modified or not.
+            (str or None): file ID if exists.
         """
-        if self.is_renamed:
-            print_warning("fromversion might have been modified, please make sure it hasn't changed.")
-            return True
+        file_id = loaded_file_data.get('id')
+        if not file_id:
+            # In integrations/scripts, the id is under 'commonfields'.
+            file_id = loaded_file_data.get('commonfields', {}).get('id')
+        if not file_id:
+            # In layout, the id is under 'layout'.
+            file_id = loaded_file_data.get('layout', {}).get('id')
+        return file_id
 
-        if not change_string:
-            change_string = run_command("git diff HEAD {0}".format(self.file_path))
+    def is_file_id_without_slashes(self):
+        # type: () -> bool
+        """Check if the ID of the file contains any slashes ('/').
 
-        is_added_from_version = re.search(r"\+([ ]+)?fromversion: .*", change_string)
-        is_added_from_version_secondary = re.search(r"\+([ ]+)?\"fromVersion\": .*", change_string)
+        Returns:
+            bool. Whether the file's ID contains slashes or not.
+        """
+        file_id = self.get_file_id_from_loaded_file_data(self.current_file)
+        if file_id and '/' in file_id:
+            self.is_valid = False
+            print_error(Errors.file_id_contains_slashes())
+            return False
+        return True
 
-        if is_added_from_version or is_added_from_version_secondary:
-            print_error("You've added fromversion to an existing file in the system, this is not allowed, please undo. "
-                        "the file was {}.".format(self.file_path))
-            self._is_valid = False
+    def is_id_modified(self, change_string=None):
```

Comment:
while we are refactoring everything here, change the verification to be based on `self._old_file` instead of `git diff`

should be part of the `BaseValidator`